var/spack/repos/builtin/packages/libcatalyst/package.py

```diff
@@ -36,3 +46,28 @@ def cmake_args(self):
         ]
 
         return args
+
+    def smoke_test(self):
+        spec = self.spec
+        testdir = "smoke_test_build"
+
+        with working_dir(testdir, create=True):
+            cmake = Executable(spec["cmake"].prefix.bin.cmake)
+            cmakeExampleDir = join_path(self.stage.source_path, "examples")
+            catalystPath = join_path(spec["libcatalyst"].prefix.lib.cmake, "catalyst-2.0")
+            cmakeArgs = [
+                cmakeExampleDir,
+                "-Dcatalyst_DIR=" + catalystPath,
+                "-DBUILD_SHARED_LIBS=ON",
+            ]
+            cmake(*(cmakeArgs))
+            cmake(*(["--build", "."]))
+            tty.info("Running Catalyst test")
+
+            res = subprocess.run(["adaptor0/adaptor0_test", "catalyst"])
+            assert res.returncode == 0
+
+    @on_package_attributes(run_tests=True)
+    @run_after("install")
+    def build_test(self):
+        self.smoke_test()
```

Comment:
Under the circumstances, you can move the contents of `smoke_test()` into this method.

Reply:
Done! :)